tidy3d/web/api/mode.py

```diff
@@ -137,6 +145,106 @@ def run(
+    task_name : str
+        Base name for tasks. Each task in the batch will have a unique index appended to this base name.
+    folder_name : str
+        Name of the folder where tasks are stored on the server's web UI.
+    results_files : List[str], optional
+        List of file paths where the results for each ModeSolver should be downloaded. If None, a default path based on the folder name and index is used.
+    verbose : bool
+        If True, displays a progress bar. If False, runs silently.
+    max_workers : int
+        Maximum number of concurrent workers to use for processing the batch of simulations.
+    max_retries : int
+        Maximum number of retries for each simulation in case of failure before giving up.
+    retry_delay : int
+        Delay in seconds between retries when a simulation fails.
+    progress_callback_upload : Callable[[float], None], optional
+        Optional callback function called when uploading file with ``bytes_in_chunk`` as argument.
+    progress_callback_download : Callable[[float], None], optional
+        Optional callback function called when downloading file with ``bytes_in_chunk`` as argument.
+
+
+    Returns
+    -------
+    List[ModeSolverData]
+        A list of ModeSolverData objects containing the results from each simulation in the batch. ``None`` is placed in the list for simulations that fail after all retries.
+    """
+    console = get_logging_console()
+
+    # Check type to make sure the user is submitting a list of mode solvers
+    if not all(isinstance(x, ModeSolver) for x in mode_solvers):
+        log.error("[red]Stopped running a batch of mode solver simulations. Please provide a list of mode solvers of type [cyan]`td.plugins.mode.mode_solver.ModeSolver` [red]to run a batch of mode solver simulations.")
+        return
+    
+    num_mode_solvers = len(mode_solvers)
+
+    # Result files where the data is stored after downloading
+    if results_files is None:
+        results_files = [f"{folder_name}/mode_solver_results_{i}.hdf5" for i in range(num_mode_solvers)]
+
+    results = [None] * num_mode_solvers 
+    
+    # Handling simulation to make sure there are more than one tries if simulation produces some error
+    def handle_mode_solver(index, retries=0):
+        try:
+            # We'll create a separate progress bar for batch simulations so we'll set verbose=False here for individual runs
+            result = run(
+                        mode_solver = mode_solvers[index], 
+                         task_name = f"{task_name}_{index}", 
+                         mode_solver_name = f"mode_solver_batch_{index}", 
+                         folder_name = folder_name, 
+                         results_file = results_files[index], 
+                         verbose = False, 
+                         progress_callback_upload = progress_callback_upload, 
+                         progress_callback_download = progress_callback_download
```

Comment:
the style guide we follow usually doesn't put spaces between the `=` in kwargs. You can automatically format the code using the `black` tool. See our notion page in the EM wiki. In fact the tests won't pass until the code is unchanged under `black` and `ruff`. Make sure you have the version prescribed in the `pyproject.toml` 

https://www.notion.so/flexcompute/Python-Style-Guide-afac5dbbf949463e89c18fe39b7453c9